CSharpMinifier/Rewriters/CSharpRewriter.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Rename;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace CSharpMinifier.Rewriters
+{
+    class CSharpRewriter : CSharpSyntaxRewriter
+    {
+        private SemanticModel _semanticModel;
+        private AdhocWorkspace _workspace;
+        private MinifierOptions _options;
+        private IdentifierGenerator _identifierGenerator; 
+
+        public CSharpRewriter(MinifierOptions options = null, bool visitIntoStructuredTrivia = true) : base(visitIntoStructuredTrivia)
+        {
+            _options = options ?? new MinifierOptions();
+            _identifierGenerator = new IdentifierGenerator();
+            //Add cause MSBuild does not work without it
+            var _ = typeof(Microsoft.CodeAnalysis.CSharp.Formatting.CSharpFormattingOptions);
+        }
+
+        public SyntaxNode VisitAndRename(SyntaxNode node)
+        {
+            node = Visit(node);
+            return RenameAll(node);
+        }
+
+        private SyntaxNode RenameAll(SyntaxNode node)
+        {
+            var mscorlib = MetadataReference.CreateFromFile(typeof(object).Assembly.Location);
+            _workspace = new AdhocWorkspace();
+            var project = _workspace.AddProject("MinifierProject", LanguageNames.CSharp);
+            project = project.AddMetadataReference(mscorlib);
+            _workspace.TryApplyChanges(project.Solution);
+            var source = node.GetText();
+            var document = _workspace.AddDocument(project.Id, "Doc", source);
+            _semanticModel = document.GetSemanticModelAsync().Result;
+            var newNode = _semanticModel.SyntaxTree.GetRoot();
+            foreach (KeyValuePair<VariableDeclaratorSyntax, string> variable in _identifierGenerator.RenamedVariables)
+            {
+                var nodeToSearch = newNode.DescendantNodes()
+                    .OfType<VariableDeclaratorSyntax>().FirstOrDefault(x => x.Identifier.ValueText.Equals(variable.Key.Identifier.ValueText));
```

Comment:
Type of Identifier `SyntaxToken` implements `IEquatable<SyntaxToken>` interface. Is it possible to use Equals method from there?